packages/core/src/providers/azure-devops.ts

```diff
@@ -0,0 +1,174 @@
+ *
+ * :::warning
+ * You are required to use HTTPS even for the localhost
+ * :::
+ *
+ * :::warning
+ * You will have to delete and create a new application to change the scopes later
+ * :::
+ *
+ * The following data is relevant for the next step:
+ *
+ * - App ID
+ * - Client Secret (after clicking the ‘Show’ button, ignore App Secret entry above it)
+ * - Authorized Scopes
+ *
+ * ### Set up the environment variables
+ *
+ * In `.env.local` create the following entries:
+ *
+ * ```
+ * AZURE_DEVOPS_APP_ID=<copy App ID value here>
+ * AZURE_DEVOPS_CLIENT_SECRET=<copy generated client secret value here>
+ * AZURE_DEVOPS_SCOPE=<copy space separated Authorized Scopes list here>
+ * ```
+ *
+ * ## Example
+ *
+ * ```js title="pages/api/auth/[...nextauth].js"
+ * import AzureDevOpsProvider from "next-auth/providers/azure-devops"
+ * ...
+ * providers: [
+ *   AzureDevOpsProvider({
+ *     clientId: process.env.AZURE_DEVOPS_APP_ID,
+ *     clientSecret: process.env.AZURE_DEVOPS_CLIENT_SECRET,
+ *     scope: process.env.AZURE_DEVOPS_SCOPE,
+ *   }),
+ * ]
+ * ...
+ * ```
+ *
+ * ### Refresh token rotation
+ *
+ * Use the [main guide](/guides/basics/refresh-token-rotation) as your starting point with the following considerations:
+ *
+ * ```js title="pages/api/auth/[...nextauth].js"
+ * async jwt({ token, user, account }) {
+ *   ...
+ *   // The token has an absolute expiration time
+ *   const accessTokenExpires = account.expires_at * 1000
+ *   ...
+ * }
+ *
+ * async function refreshAccessToken(token) {
+ *   ...
+ *   const response = await fetch(
+ *     "https://app.vssps.visualstudio.com/oauth2/token",
+ *     {
+ *       headers: { "Content-Type": "application/x-www-form-urlencoded" },
+ *       method: "POST",
+ *       body: new URLSearchParams({
+ *         client_assertion_type:
+ *           "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+ *         client_assertion: AZURE_DEVOPS_CLIENT_SECRET,
+ *         grant_type: "refresh_token",
+ *         assertion: token.refreshToken,
+ *         redirect_uri:
+ *           process.env.NEXTAUTH_URL + "/api/auth/callback/azure-devops",
+ *       }),
+ *     }
+ *   )
+ *   ...
+ *   // The refreshed token comes with a relative expiration time
+ *   const accessTokenExpires = Date.now() + newToken.expires_in * 1000
+ *   ...
+ * }
+ * ```
+ */
+export default function AzureDevOpsProvider<P extends AzureDevOpsProfile>(
+  options: OAuthUserConfig<P> & {
+    /**
+     * https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes
+     * @default vso.profile
+     */
+    scope?: string;
+  }
+): OAuthConfig<P> {
+  const scope = options.scope ?? 'vso.profile';
+
+  return {
+    id: 'azure-devops',
+    name: 'Azure DevOps',
+    type: 'oauth',
+
+    authorization: {
+      url: 'https://app.vssps.visualstudio.com/oauth2/authorize',
+      params: { response_type: 'Assertion', scope }
+    },
+
+    token: {
```

Comment:
The example app works, could you link the example repo? I find it a bit weird because in the code, you're missing `url` for both `token` and `userinfo` config, so Auth.js is complaining: 
```

next-auth-app:dev: [auth][error][InvalidEndpoints]: Provider "azure-devops" is missing both `issuer` and `token` endpoint config. At least one of them is required.. Read more at https://errors.authjs.dev#invalidendpoints
next-auth-app:dev:     at assertConfig (webpack-internal:///(middleware)/../../../packages/core/lib/assert.js:76:24)
next-auth-app:dev:     at Auth (webpack-internal:///(middleware)/../../../packages/core/index.js:82:89)
next-auth-app:dev:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Could you add back `url` to `token` and `userinfo`?